### PR TITLE
Np 45083 upsert nvi candidate

### PR DESCRIPTION
**Pull request by monaullah**
Splitting Jira task (https://unit.atlassian.net/browse/NP-45083) into several PRs to keep them small. This is nr 2

UpsertNviCandidateHandler consumes messages on sqs queue "NewCandidateQueue"
- If the candidate does not already exist in DB, create new candidate with pending approval statuses and save.
- Faking NviCandidateRepository util DB Tech is chosen and implemented

### CODE REVIEW

File: event-handlers/src/main/java/handlers/UpsertNviCandidateHandler.java [hunk 1]
```diff
@@ -7,15 +7,32 @@
...
+import nva.commons.core.paths.UriWrapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class UpsertNviCandidateHandler implements RequestHandler<SQSEvent, Void> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(UpsertNviCandidateHandler.class);
 
+    private static final Environment ENVIRONMENT = new Environment();
+    private static final String PUBLICATION_API_HOST = ENVIRONMENT.readEnv("PUBLICATION_API_HOST");
```
Comment (truhacevkir) [inline event-handlers/src/main/java/handlers/UpsertNviCandidateHandler.java:26]: API_HOST?

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviCandidateService.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    public Optional<Candidate> getCandidateByPublicationId(URI publicationId) {
+        return nviCandidateRepository.findByPublicationId(publicationId.toString());
+    }
+
+    public boolean exists(URI publicationId) {
+        return nviCandidateRepository.findByPublicationId(publicationId.toString()).isPresent();
+    }
+
+    public void createCandidateWithPendingInstitutionApprovals(URI publicationId, List<URI> institutionUri) {
```
Comment (truhacevkir) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviCandidateService.java:27]: Will it be possible to create Approved NviCandidate? I guess no, maybe rename to createCandidate?

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/NviCandidateServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                                    .build();
+
+        nviCandidateService.createCandidateWithPendingInstitutionApprovals(publicationId, List.of(institutionId));
+
+        assertThat(fakeNviCandidateRepository.findByPublicationId(publicationId.toString()),
+                   is(equalTo(Optional.of(expectedCandidate))));
+    }
+
+    private URI generatePublicationId(UUID identifier) {
+        return UriWrapper.fromHost(PUBLICATION_API_HOST)
```
Comment (truhacevkir) [inline nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/NviCandidateServiceTest.java]: API_HOST

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviCandidateService.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import no.sikt.nva.nvi.common.NviCandidateRepository;
+import no.sikt.nva.nvi.common.model.business.ApprovalStatus;
+import no.sikt.nva.nvi.common.model.business.Candidate;
+import no.sikt.nva.nvi.common.model.business.Status;
+
+public class NviCandidateService {
```
Comment (truhacevkir) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviCandidateService.java:11]: What about nviService?

**Review by truhacevkir — COMMENTED** [review]
Is it nviService we are going to use for database transactions? or the future service that will implement NviCandidateRepository? 
If it is nviService, it seems correctly for nviService to implement NviCandidateRepository? or am I misunderstanding something?

**Comment by monaullah** [comment]
@truhacevkir NviCandidateRepository will contain the DB transactions, and NviService will contain the business logic. That's why I'm faking NviCandidateRepository atm

**Review by truhacevkir — APPROVED** [review]

**Pull request merged**